src/merkle.rs

```diff
@@ -551,7 +551,7 @@ impl<
             // Total number of leafs in the compound tree is the combined leafs total of all subtrees.
             let leafs = trees.iter().fold(0, |leafs, mt| leafs + mt.leafs());
             // Total length of the compound tree is the combined length of all subtrees plus the root.
-            let len = trees.iter().fold(0, |len, mt| len + mt.len()) + 1;
+            let len = trees[0].len() * trees.len() + 1;
```

Comment:
Could you please add a comment here, something like "All trees are guaranteed to have the same length, it's asserted above." (and the same at the change below).

Reply:
Done.